part-2.py

```diff
@@ -4,13 +4,43 @@
 # the appropriate comment.
 
 # search
-
+def search(array, query):
+    if len(array) == 0:
+        return False
+    if array[0] == query:
+        return True
+    else:
+        return search(array[1:], query)
 
 
 # is_palindrome
+def is_palindrome(text):
+    if len(text) == 1 or len(text) == 0:
+        return True
 
+    if text[0] != text[-1]:
+        return False
 
+    return is_palindrome(text[1:-1])
 
-# digit_match
 
+# digit_match
+def digit_match(int_1, int_2):
+    str_1 = str(int_1)
+    str_2 = str(int_2)
+
+    if str_1[-1] != str_2[-1]: 
+        str_1 = str_1[:-1]
+        str_2 = str_2[:-1]
+
+        return digit_match(str_1[:-1], str_2[:-1])
+  
+    else:
+        str_1 = str_1
+        str_2 = str_2
+        return digit_match(str_1[:-1], str_2[:-1])
```

Comment:
Sure! We can definitely talk about this. Let me give you some feedback here, too.

So, you are stringifying the two numbers and starting form the back to match the ones position with the ones position and so on.

If they are equal, we could set up a counter to increment. Then whether or not it is equal or not, we call the function again recursively, incrementing the strings for the next call.

Then when you have done through each character int he strings, return the counter.

This solution here doesn't pass the Learn tests